blocks/library/gallery/block.js

```diff
@@ -56,15 +56,15 @@ class GalleryBlock extends Component {
 	}
 
 	onSelectImage( index ) {
-		return ( event ) => {
-			// ignore clicks in the editable caption.
-			// Without this logic, text operations like selection, select / unselects the images.
-			if ( event.target.tagName === 'FIGCAPTION' ) {
+		return () => {
+			// if the image is already selected do nothing
+			if ( this.state.selectedImage === index ) {
```

Comment:
With this added, the toolbar will remain at the caption field when clicking the related image. Never reaches the `setFocus` call. Here's what I suggested: https://github.com/WordPress/gutenberg/pull/4199#discussion_r162589825.

Reply:
Ah ok, thank you @iseulde I got the idea of the behavior you are suggesting and I totally agree with it.
I updated the code to follow that behavior although in this behavior we still have to check if the clicks are in FIGCAPTION element because otherwise each time we do actions in the RichText we will lose the focus of it.